BDS/hydro_bds_edge_state_2D.cpp

```diff
@@ -401,7 +374,7 @@ BDS::ComputeConc (Box const& bx,
         // set edge values equal to the ghost cell value since they store the physical condition on the boundary
         if ( i==dlo.x && lo_x_physbc ) {
             sedgex(i,j,k,icomp) = s(i-1,j,k,icomp);
-            if (is_velocity && icomp == XVEL && (bc.lo(0) == BCType::foextrap ||  bc.lo(0) == BCType::hoextrap) ) {
+            if (is_velocity && icomp == XVEL && (h_bc.lo(0) == BCType::foextrap || h_bc.lo(0) == BCType::hoextrap) ) {
```

Comment:
Don't we want to use the device BC array inside `ParralelFor` and not the host version?

Reply:
Yeah that makes sense.  The code actually runs on the GPU like this; do you understand why we need to make this distinction here?

Reply:
At line 112 `auto h_bc = h_bcrec.data()[icomp];`  So `h_bc` here at the new line 377 is a copy (i.e., lambda captured by value) of that earlier h_bc.  I could not see where `bc` in the old code comes from by just looking this github page.  Is that a member of the BDS class?  If so, it's actually `this->bc` to the compiler.  That does not work, because of the capture of `this`.

Reply:
Oh. `bc` is a local variable, `auto bc = pbc[icomp]`.  So both `bc` and `h_bc` work.

Reply:
Got it, so if I had tried to use pbc[icomp] directly it would NOT work on the GPU, right?

Reply:
I reverted it anyway

Reply:
A change was made recently because what was there did not work for all cases. See PR #101 https://github.com/AMReX-Codes/AMReX-Hydro/pull/101

I guess it's not clear from the code, since pbc is passed as a pointer, but the idea (and what incflo and IAMR do) is that pbc is on the device and  h_bc is on the host.

Reply:
Right, but once `h_bc` is captured onto the device, it's on the device.  The `h_bc` in the kernel is actually a device copy of the host variable.  I am not saying `h_bc` instead of `bc` should be used in the kernel.  Either way is fine.  I am just saying `h_bc` in the kernel is technically a device variable despite `h_` in its name.

Reply:
@marchdf Can you shed some light on what precipitated PR #101? I was under the impression that it actually failed for a case you tried running.

Reply:
```
auto bc = pbc[icomp];
```
I think that was because the line above ascess device data (pointed to by pbc) on the host.

Reply:
One example of the change in that pr
```
    auto bc = pbc[icomp];
    auto h_bc = h_bcrec.data()[icomp];
```
This is host code.

Reply:
@WeiqunZhang is right. I was trying to make it clear what was accessed on host and device because things would fail if use_managed = 0. But it looks like I left some implicit copies in there? Sorry